spec/unit/collect_spec.rb

```diff
@@ -4,6 +4,50 @@
 
   subject(:prompt) { TTY::TestPrompt.new }
 
+  def collect(&block)
+    prompt = subject
+    count = 0
+
+    result = prompt.collect do
+      while prompt.yes?("continue?")
+        instance_eval(&block)
+        count += 1
+      end
+    end
+
+    result[:count] = count
+    result
+  end
+
+  context "when receiving multiple answers" do
+    let(:colors) { %w(red blue yellow) }
+
+    before do
+      subject.input << "y\r" + colors.join("\ry\r") + "\rn\r"
+      subject.input.rewind
+    end
+
```

Comment:
I would add test case that has more than one key with multiple values and a key with a single value to ensure that is no mutation of wrong keys.